spec/doorkeeper-jwt/doorkeeper-jwt_spec.rb

```diff
@@ -133,11 +133,55 @@
       end
 
       token = Doorkeeper::JWT.generate({})
-      decoded_token = ::JWT.decode(token, secret_key, "RS512")
+      decoded_token = ::JWT.decode(token, secret_key, true, {algorithm: "RS512"})
       expect(decoded_token[0]).to be_a(Hash)
       expect(decoded_token[0]["foo"]).to eq "bar"
       expect(decoded_token[1]["typ"]).to eq "JWT"
       expect(decoded_token[1]["alg"]).to eq "RS512"
     end
+    
+    it "creates a signed JWT token encrypted with an ECDSA key from a file" do
+      Doorkeeper::JWT.configure do
+        token_payload do
+          {
+            foo: "bar"
+          }
+        end
+        secret_key_path "spec/support/512key.pem"
+        encryption_method :es512
+      end
+
+      token = Doorkeeper::JWT.generate({})
+      secret_key = OpenSSL::PKey::EC.new File.read("spec/support/512key_pub.pem")
+      decoded_token = ::JWT.decode(token, secret_key, true, {algorithm: "ES512"})
+      expect(decoded_token[0]).to be_a(Hash)
+      expect(decoded_token[0]["foo"]).to eq "bar"
+      expect(decoded_token[1]["typ"]).to eq "JWT"
+      expect(decoded_token[1]["alg"]).to eq "ES512"
+    end
+
+    it "creates a signed JWT token encrypted with an ECDSA key from a string" do
+      secret_key = OpenSSL::PKey::EC.new('secp521r1')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.